substrate-node/pallets/pallet-smart-contract/src/lib.rs

```diff
@@ -2365,7 +2375,6 @@ impl<T: Config> ChangeNode<LocationOf<T>, InterfaceOf<T>, SerialNumberOf<T>> for
                     &types::ContractState::Deleted(types::Cause::CanceledByUser),
                 );
                 let _ = Self::bill_contract(node_contract_id);
-                Self::remove_contract(node_contract_id);
```

Comment:
Remove contract not needed here because it already done inside `bill_contract()`

Reply:
Indeed